### PR TITLE
Hooks to sync to allow doing an action before or after entire sync step

### PR DESCRIPTION
**Pull request by mrGrazy**
### Sync callbacks 

You may want to attach some application logic to happen before or after a sync run, for example, to update caches or something. 

Callbacks can be registered using:

```ruby
Consyncful::Sync.before_run do
  #do something before the run
end
```

```ruby
Consyncful::Sync.after_run do |updated_ids|
  #invalidate cache for updated_ids, or something
end
```

### CODE REVIEW

File: lib/consyncful/sync.rb [hunk 1]
```diff
@@ -52,6 +60,12 @@ def load_all_models
       Rails.application.eager_load!
     end
 
+    def update_run(next_url)
```
Comment (mrGrazy) [inline lib/consyncful/sync.rb:63]: Extract method from `run` to reduce line count

File: README.md [hunk 2]
```diff
@@ -134,6 +134,24 @@ Because all contentful models are stored as polymorphic subtypes of Consyncful::
...
+Consyncful::Sync.before_run do
+  #do something before the run
+end
+```
+
+```ruby
+Consyncful::Sync.after_run do |updated_ids|
+  #invalidate cache for updated_ids, or something
+end
+```
```
Comment (invke) [inline README.md:153]: You don't have a space after the `#` in your comments. 🤷‍♂️ 🤸‍♂️

File: lib/consyncful/sync.rb (hunk 1 above)
Comment (hapiben) [reply lib/consyncful/sync.rb:63]: Neat.

**Review by invke — APPROVED** [review]
Noooooice. I liked your usage of refactoring terminology🥇

**Review by hapiben — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mrGrazy) [reply README.md:153]: True.. rubocop doesn't work in markdown codeblocks

**Review by invke — COMMENTED** [review]
Oh, I just realised, you'll need to bump the version right @mrGrazy?

**Pull request merged**